staging/src/k8s.io/client-go/util/jsonpath/jsonpath.go

```diff
@@ -276,6 +276,27 @@ func (j *JSONPath) evalIdentifier(input []reflect.Value, node *IdentifierNode) (
 	return results, nil
 }
 
+func (j *JSONPath) evalMap(input []reflect.Value, hasAsterisk bool) ([]reflect.Value, error) {
+	result := []reflect.Value{}
+	if !hasAsterisk {
+		return nil, fmt.Errorf("invalid jsonpath for objects. Index should be an asterisk when applied over an object")
```

Comment:
why is this the case? reading https://goessner.net/articles/JsonPath/index.html#e2 I'd expect indexing to be valid for maps:

> JSONPath expressions can use the dot–notation
>
> $.store.book[0].title
>
> or the bracket–notation
>
> $['store']['book'][0]['title']

Reply:
that case is done for the following:

suppose we have an object,
```
{
	"store": {
		"book": {
			"category": "reference",
			"author": "Nigel Rees",
			"title": "Sayings of the Century",
			"price": 8.95
		},
		"bicycle": {
			"color": "red",
			"price": 19.95
		}
	}
}
```

so, the allowed jsonpaths (docs) are:
1. `$.store[*]`
2. `$.store.book[*]`
etc.

but,
1. `$.store[0]` is not allowed.
2. `$.store.book[*]` is not allowed.

In the link you mentioned, the case would work according to the code in the PR,
1. `$.store.book[0].title` will work since book is a slice and not a map. Infact, `book` is a slice of maps. Hence, `$.store.book[0].title`, `$.store.book[0][*]` is allowed, but `$.store.book[0][0]` is not. **This declared `fmt.Errorf()` is done to catch the `$.store.book[0][0]` case so that the there can be no digit in maps. Only `*` is allowed in case of maps.**

Reply:
> 1. Only `*` is allowed in case of maps.

That does not match the description in https://goessner.net/articles/JsonPath/index.html#e2

Shouldn't specific string keys be allowed for maps?

Reply:
Sorry. What I meant by that is, in maps, asterisks are allowed inside `[ ]` and not integers. Of course, string paths are allowed as well.